src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -0,0 +1,492 @@
+
+                    // Local Functions.
+                    bool HasSyntaxErrors()
+                    {
+                        foreach (var operationBlock in context.OperationBlocks)
+                        {
+                            if (operationBlock.SemanticModel.GetSyntaxDiagnostics(operationBlock.Syntax.Span, context.CancellationToken).HasAnyErrors())
+                            {
+                                return true;
+                            }
+                        }
+
+                        return false;
+                    }
+                }
+
+                private void AnalyzeExpressionStatement(OperationAnalysisContext context)
+                {
+                    if (_options.UnusedValueExpressionStatementSeverity == ReportDiagnostic.Suppress)
+                    {
+                        return;
+                    }
+
+                    var expressionStatement = (IExpressionStatementOperation)context.Operation;
+                    var value = expressionStatement.Operation;
+
+                    // Bail out cases for report unused expression value:
+
+                    //  1. Null type and void returning method invocations: no value being dropped here.
+                    if (value.Type == null ||
+                        value.Type.SpecialType == SpecialType.System_Void)
+                    {
+                        return;
+                    }
+
+                    //  2. Bail out for syntax error (constant expressions) and semantic error (invalid operation) cases.
+                    if (value.ConstantValue.HasValue ||
+                        value is IInvalidOperation)
+                    {
+                        return;
+                    }
+
+                    //  3. Assignments, increment/decrement operations: value is actually being assigned.
+                    if (value is IAssignmentOperation ||
+                        value is IIncrementOrDecrementOperation)
+                    {
+                        return;
+                    }
+
+                    //  4. Bool returning method invocations: these are extremely noisy as large number of bool 
+                    //     returning methods return the status of the operation.
+                    //     In future, if we feel noise level is fine, we can consider removing this check.
+                    if (value.Type.SpecialType == SpecialType.System_Boolean)
+                    {
+                        return;
+                    }
+
+                    //  5. If expression statement and its underlying expression have differing first tokens that likely indicates
+                    //     an explicit discard. For example, VB call statement is used to explicitly ignore the value returned by
+                    //     an invocation by prefixing the invocation with keyword "Call".
+                    if (value.Syntax.GetFirstToken() != expressionStatement.Syntax.GetFirstToken())
```

Comment:
that's a large stretch.  If we want to handle VB call expressions, we should have a method that checks for that, or otherwise defers to each language for language specific checks.